### ISSUE TITLE
extra_feature_edges does not seem to have any effect

### ISSUE THREAD

**Issue by lahwaacz** [nschloe/pygalmesh#182]
I tried the example for [domain combination](https://github.com/nschloe/pygalmesh#domain-combinations) with extra feature edges (i.e. the second code block in the linked section). However, I've got this mesh:

[![balls-difference.png](https://i.postimg.cc/hj12J2HM/balls-difference.png)](https://postimg.cc/2b3QM7jB)

Also the edge length seems to be non-uniform along the circle where the spheres intersected. Any idea what went wrong?

**Comment by nschloe** [comment]
I'd have to look at it in more detail. I don't have much time these days, so don't hold your breath.

**Comment by lahwaacz** [comment]
No problem, take your time. I might dig into it myself when I find more time.

**Comment by lahwaacz** [comment]
I found the problem. In [the example](https://github.com/nschloe/pygalmesh#domain-combinations), the feature points are sampled based on angles
```py
alpha = np.linspace(0.0, 2 * np.pi, n + 1)
```
The first element of this array is 0, but the last element is -2.12115048e-16 due to rounding. This has the weird consequence in CGAL effectively not preserving features – probably because the smallest distance between feature points was about 1e-16, but `max_edge_size_at_feature_edges` was set to 0.15. When I overwrote the last angle with the first, `alpha[-1] = alpha[0]`, I got the expected result with preserved features.
